layer/layer.go

```diff
@@ -241,7 +237,7 @@ type MetadataStore interface {
 	GetDiffID(ChainID) (DiffID, error)
 	GetCacheID(ChainID) (string, error)
 	GetDescriptor(ChainID) (distribution.Descriptor, error)
-	GetOS(ChainID) (OS, error)
+	GetOS(ChainID) (string, error)
```

Comment:
Where is this used? If you just need to keep the mapping then this doesn't need to be exposed in the API.

Reply:
It's used in `loadLayer`. It's a safety precaution when loading a layer from disk to prevent a layer from a "foreign" non-matching operating system. I should be able to remove it from the interface and keep as an internal function.